### PR TITLE
@l2succes => Toolchain updates to match Reaction

### PR DESCRIPTION
**Pull request by eessex**
- Pins node to 8.12.0 
- Package updates to bring Babel up to 7.0.0
- Updates Reaction to latest

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,26 +30,15 @@
...
-    "@babel/core": "7.0.0-rc.2",
-    "@babel/node": "7.0.0-rc.2",
-    "@babel/plugin-proposal-class-properties": "7.0.0-rc.2",
-    "@babel/plugin-proposal-decorators": "7.0.0-rc.2",
-    "@babel/plugin-transform-runtime": "7.0.0-rc.2",
-    "@babel/preset-env": "7.0.0-rc.2",
-    "@babel/preset-react": "7.0.0-rc.2",
-    "@babel/preset-typescript": "7.0.0-rc.2",
-    "@babel/register": "7.0.0-rc.2",
-    "@babel/runtime": "7.0.0-rc.2",
```
Comment (damassi) [inline package.json]: If we're not fully compiling out our server-side CoffeeScript and other ES6 / TypeScript before sending to prod then moving these modules to `devDeps` will fail during deploy. (Things like `coffee-script/register` and `@babel/register` mean that code is transformed during runtime (and then cached).

File: .babelrc [hunk 2]
```diff
@@ -3,7 +3,8 @@
     ["@babel/env", {
       "targets": {
         "browsers": ["last 2 versions"]
-      }
+      },
+      "useBuiltIns": "usage"
```
Comment (damassi) [inline .babelrc]: Can you do some investigating around `useBuiltIns: "usage"` and check to see if `@babel/plugin-transform-runtime` is still required? I _believe_ this supersedes it.

Comment (eessex) [reply .babelrc]: Yeah looks like it runs fine without `@babel/plugin-transform-runtime`

File: package.json (hunk 1 above)
Comment (eessex) [reply package.json]: Are we not fully compiling before we deploy? I thought that was what the `yarn assets` step is for.

Comment (damassi) [reply package.json]: Thats only for the client-side

File: .babelrc [hunk 3]
```diff
@@ -3,16 +3,17 @@
...
     }],
     "@babel/react",
     "@babel/preset-typescript"
   ],
   "plugins": [
     ["@babel/plugin-proposal-decorators", { "legacy": true }],
     "@babel/plugin-proposal-class-properties",
-    "@babel/plugin-transform-runtime",
     "inline-react-svg",
+    "lodash",
```
Comment (l2succes) [inline .babelrc]: I think it's safe to remove lodash here

File: package.json (hunk 1 above)
Comment (eessex) [reply package.json]: Sounds good, put these back in `dependencies`

File: .babelrc (hunk 3 above)
Comment (damassi) [reply .babelrc]: Even though `underscore` is the official utility lib it's being used all over (same in force).

Comment (damassi) [reply .babelrc]: I wonder if there's some kind of codemod we could run to swap it out

Comment (eessex) [reply .babelrc]: You mean remove underscore? I'd been encouraged to switch to `lodash` wherever possible.

Comment (damassi) [reply .babelrc]: Yup remaining underscore usage and replace with lodash using the codemod, would automate all the work away.

Comment (damassi) [reply .babelrc]: Thinking further though, that codemod won't work with coffeescript /shrug

**Pull request merged**